[Term Entry] PyTorch Tensor Operations: .randint() (#6313)
Codecademy/docs: content/pytorch/concepts/tensor-operations/terms/randint/randint.md

---
Title: '.randint()'
Description: 'Returns a tensor filled with random integers generated uniformly between specified bounds.'
Subjects:
  - 'Computer Science'
  - 'Machine Learning'
Tags:
  - 'Functions'

>   - 'PyTorch'
>   - 'Tensor'
>   - 'Random'

Reviewer: ```suggestion
  - 'PyTorch'
  - 'Random'
  - 'Tensor'
```

CatalogContent:
  - 'learn-python-3'
  - 'paths/machine-learning'
---

> **`.randint()`** is a function in [PyTorch](https://www.codecademy.com/resources/docs/pytorch) that generates [tensors](https://www.codecademy.com/resources/docs/pytorch/tensors) filled with random integers. It creates a tensor filled with random integers generated uniformly between a lower bound (inclusive) and an upper bound (exclusive). This function is particularly useful when you need tensors with random integer values within a specific range for various machine learning tasks.

Reviewer: The first and second lines in this paragraph almost convey the same thing. It can be written in this way
```suggestion
**`.randint()`** is a function in [PyTorch](https://www.codecademy.com/resources/docs/pytorch) that generates [tensors](https://www.codecademy.com/resources/docs/pytorch/tensors) filled with random integers sampled uniformly between a specified lower bound (inclusive) and an upper bound (exclusive) This function is particularly useful when you need tensors with random integer values within a specific range for various machine learning tasks.
```

        [0., 0., 0., 1., 0.],
        [1., 1., 0., 1., 1.],
        [1., 0., 0., 0., 0.]])
```

## Example 3: Creating random dice rolls

This example demonstrates how to simulate dice rolls using `.randint()` to generate random integers between 1 and 6:

```py
import torch

# Set seed for reproducibility
torch.manual_seed(123)

# Simulate rolling a single die 10 times
single_die = torch.randint(1, 7, (10,))
print("10 dice rolls:")
print(single_die)

# Count the frequency of each number
for i in range(1, 7):
  count = (single_die == i).sum().item()
  print(f"Number {i} appeared {count} times")

# Simulate rolling 5 dice at once
dice_rolls = torch.randint(1, 7, (5,))
print("\n5 dice rolled simultaneously:")
print(dice_rolls)

# Calculate the sum of the dice
total = dice_rolls.sum().item()
print(f"Sum of all dice: {total}")
```

This example results in the following output:

```shell
10 dice rolls:
tensor([4, 2, 6, 1, 2, 6, 5, 2, 2, 5])
Number 1 appeared 1 times
Number 2 appeared 4 times
Number 3 appeared 0 times
Number 4 appeared 1 times
Number 5 appeared 2 times
Number 6 appeared 2 times

5 dice rolled simultaneously:
tensor([1, 3, 4, 6, 3])
Sum of all dice: 17
```

> To learn more about other tensor operations, visit the [PyTorch Tensor Operations](https://www.codecademy.com/resources/docs/pytorch/tensor-operations) documentation.

Reviewer: Bcaklink should be to a relevant course

> To explore more, check out this [Intro to PyTorch and Neural Networks](https://www.codecademy.com/enrolled/courses/intro-to-py-torch-and-neural-networks) course, where you’ll learn to create, train, and test neural networks.

Reviewer: Avoid using first and second-person pronouns, i.e. "I", "we", and "you".
```suggestion
To learn more about how to create, train, and test neural networks, check out this [Intro to PyTorch and Neural Networks](https://www.codecademy.com/enrolled/courses/intro-to-py-torch-and-neural-networks) course on Codecademy.
```
